ios/storage.md

```diff
@@ -161,317 +161,620 @@ func downloadData() {
+   for task in downloadTasks! {
+       task.setCompletionHandler(completionHandler!)
+       task.setProgressBlock(progressBlock!)
     }
 }
+
+
+let multiPartUploadTasks = transferUtility.getMultiPartUploadTasks().result
+    for task in multiPartUploadTasks! {
+        task.setCompletionHandler(completionHandler!)
+        task.setProgressBlock(progressBlock!)
+
+ }
+```
+
+### Transfer with Object Metadata
+
+The `AWSS3TransferUtilityUploadExpression` and `AWSS3TransferUtilityMultiPartUploadExpression` classes contain the method `setValue:forRequestHeader` where you can pass in metadata to Amazon S3. This example demonstrates passing in the Server-side Encryption Algorithm as a request header in uploading data to S3 using MultiPart.
+
+```swift
+
+let data: Data = Data() // The data to upload
+
+let uploadExpression = AWSS3TransferUtilityMultiPartUploadExpression()
+uploadExpression.setValue("AES256", forRequestHeader: "x-amz-server-side-encryption-customer-algorithm")
+uploadExpression.progressBlock = {(task, progress) in DispatchQueue.main.async(execute: {
+        // Do something e.g. Update a progress bar.
+    })
+}
+
+let transferUtility = AWSS3TransferUtility.default()
+
+transferUtility.uploadUsingMultiPart(data:data,
+            bucket: "S3BucketName",
+            key: "S3UploadKeyName",
+            contentType: "text/plain",
+            expression: uploadExpression,
+            completionHandler: nil).continueWith { (task) -> AnyObject! in
+                if let error = task.error {
+                    print("Error: \(error.localizedDescription)")
+                }
+
+                return nil;
+            }
 ```
 
-**Response Mapping Template**
+### Transfer with Access Control List
+
+To upload a file and specify permissions for it, you can use predefined grants, also known as canned ACLs. The following code shows you how to setup a file with publicRead access using the AWSS3 client.
 
+
+```swift
+//Create a AWSS3PutObjectRequest object and setup the content, bucketname, key on it.
+//use the .acl method to specify the ACL for the file
+let s3: AWSS3 = AWSS3.default()
+
+let putObjectRequest: AWSS3PutObjectRequest! = AWSS3PutObjectRequest()
+let content = "testObjectData"
+putObjectRequest.acl = AWSS3ObjectCannedACL.publicRead
+putObjectRequest.bucket = "bucket name"
+putObjectRequest.key = "file name"
+putObjectRequest.body = content
+putObjectRequest.contentLength = content.count as NSNumber
+putObjectRequest.contentType = "text/plain";
+
+s3.putObject(putObjectRequest, completionHandler: { (putObjectOutput:AWSS3PutObjectOutput? , error: Error? ) in
+    if let output = putObjectOutput {
+        print (output)
+    }
+
+    if let error = error {
+        print (error)
+    }
+})
 ```
-$util.toJson($util.dynamodb.fromS3ObjectJson($context.source.file))
+
+### Downloading to a File
+
+```swift
+let fileURL = // The file URL of the download destination.
+
+let transferUtility = AWSS3TransferUtility.default()
```

Comment:
Isn't this redundant with the other Download examples above?

Reply:
Removed.